src/lib/utils/presence-socket.ts

```diff
@@ -0,0 +1,62 @@
+import { Socket } from 'phoenix';
+
+import { PRESENCE_URL } from './env';
+
+// And connect to the path in "lib/score_saber_realtime_web/endpoint.ex". We pass the
+// token for authentication. Read below how it should be used.
+const socket =
+   typeof window !== 'undefined' &&
+   new Socket(PRESENCE_URL, {
+      params: {}
+   });
+
+if (socket) socket.connect();
+
+export default socket;
+
+enum State {
+   Menu = 'menu',
+   InGame = 'playing'
+}
+
+/// Unix timestamp serialised to a string
+type Timestamp = string;
+
+export interface StateChangeEvent {
+   timestamp: Timestamp;
+   state: State;
+}
+
+export enum GameMode {
+   Solo = 'solo',
+   Multiplayer = 'multi',
+   Practice = 'practice'
+}
+
+export interface LiveSongInfo {
+   timestamp: Timestamp;
+   mode: GameMode;
+
+   name: string;
+   subName?: string;
+   authorName: string;
+   artist: string;
+   type: string; // Standard, Lawless, OneSaber etc
```

Comment:
enum perchance

Reply:
This is deliberately not an enum because there's a fair few mods that add additional ones